packages/aws-cdk/lib/util/get-package-manager.ts

```diff
@@ -0,0 +1,20 @@
+const pmFromUserAgent = (userAgent: string) => {
+  const pmSpec = userAgent.split(' ')[0];
+  const separatorPos = pmSpec.lastIndexOf('/');
+  const name = pmSpec.substring(0, separatorPos);
```

Comment:
magic. We need a quick comment showing example(s) of what this string is expecting